### PR TITLE
Feature/marp 315 open rest editor detail pages from within ai designer

### PR DESCRIPTION
**Pull request by nntthuy-axonivy**

### CODE REVIEW

File: marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java [hunk 1]
```diff
@@ -103,15 +105,20 @@ public ProductServiceImpl(ProductRepository productRepository, GHAxonIvyMarketRe
...
     final var searchPageable = refinePagination(language, pageable);
     var searchCriteria = new ProductSearchCriteria();
-    searchCriteria.setType(typeOption);
     searchCriteria.setListed(true);
     searchCriteria.setKeyword(keyword);
-    searchCriteria.setLanguage(Language.of(language));
-    searchCriteria.setType(typeOption);
+    if (BooleanUtils.isTrue(isRestDesigner)) {
+      searchCriteria.setType(TypeOption.CONNECTORS);
+      searchCriteria.setLanguage(Language.of(Locale.ENGLISH.toLanguageTag()));
```
Comment (nqhoan-axonivy) [inline marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java:117]: Please use Language.EN.getValue()

File: marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java [hunk 2]
```diff
@@ -216,37 +223,37 @@ private void updateLatestChangeToProductsFromGithubRepo() {
...
+        }
+        break;
+      case REMOVED:
+        result = productRepository.findByLogoUrl(product.getLogoUrl());
+        if (result != null) {
+          productRepository.deleteById(result.getId());
+        }
+        break;
+      default:
+        break;
```
Comment (nqhoan-axonivy) [inline marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java:243]: Did you change any logic from 226 to 243? If no, please revert the change for that block 226 to 243

File: marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java [hunk 3]
```diff
@@ -216,37 +223,37 @@ private void updateLatestChangeToProductsFromGithubRepo() {
...
     switch (file.getStatus()) {
-    case MODIFIED, ADDED:
-      productRepository.save(product);
-      break;
-    case REMOVED:
-      productRepository.deleteById(product.getId());
-      break;
-    default:
-      break;
+      case MODIFIED, ADDED:
```
Comment (nqhoan-axonivy) [inline marketplace-service/src/main/java/com/axonivy/market/service/impl/ProductServiceImpl.java:249]: Did you change any logic from 249 to 256? If no, please revert the change for that block 249 to 256

File: marketplace-ui/src/app/modules/product/product-card/product-card.component.html [hunk 4]
```diff
@@ -1,28 +1,42 @@
...
       width="70"
       height="70"
       [ngSrc]="product | logo"
-      [alt]="product.names | multilingualism: languageService.selectedLanguage()" />
-    <div
-      class="card__tag lh-md px-2 py-1"
-      [ngClass]="themeService.isDarkMode() ? 'text-dark' : 'text-light'">
-      {{ 'common.filter.value.' + product.type | translate }}
-    </div>
+      [alt]="
```
Comment (nqhoan-axonivy) [inline marketplace-ui/src/app/modules/product/product-card/product-card.component.html:10]: hard to read this line, please revert it to old previous version

File: marketplace-ui/src/app/modules/product/product-card/product-card.component.ts [hunk 5]
```diff
@@ -6,17 +6,26 @@ import { ThemeService } from '../../../core/services/theme/theme.service';
...
+    NgOptimizedImage
+  ],
   templateUrl: './product-card.component.html',
   styleUrl: './product-card.component.scss'
 })
 export class ProductCardComponent {
   themeService = inject(ThemeService);
   languageService = inject(LanguageService);
 
+  isProductInRestClient = inject(ProductComponent).isRestClient();
```
Comment (nqhoan-axonivy) [inline marketplace-ui/src/app/modules/product/product-card/product-card.component.ts:28]: Please rename to 'isShowInRESTClientEditor' or 'isRenderInRESTClientEditor'

File: marketplace-ui/src/app/modules/product/product.component.ts [hunk 6]
```diff
@@ -52,21 +54,35 @@ export class ProductComponent implements AfterViewInit, OnDestroy {
   criteria: Criteria = {
     search: '',
     type: TypeOption.All_TYPES,
+    isRestDesigner: false,
```
Comment (nqhoan-axonivy) [inline marketplace-ui/src/app/modules/product/product.component.ts:57]: Please chaneg the 'isRestDesigner' to 'isRESTClientEditor'

File: marketplace-ui/src/app/shared/enums/request-param.ts [hunk 7]
```diff
@@ -2,5 +2,6 @@ export enum RequestParam {
   TYPE = 'type',
   KEYWORD = 'keyword',
   SORT = 'sort',
-  LANGUAGE = 'language'
-}
\ No newline at end of file
+  LANGUAGE = 'language',
+  IS_REST_DESIGNER = 'isRestDesigner'
```
Comment (nqhoan-axonivy) [inline marketplace-ui/src/app/shared/enums/request-param.ts:6]: please rename the 'isRestDesigner' to is 'isRESTClientEditor'

File: marketplace-ui/src/app/shared/models/criteria.model.ts [hunk 8]
```diff
@@ -7,5 +7,6 @@ export interface Criteria {
   sort: SortOption | null;
   type: TypeOption | null;
   language: Language;
+  isRestDesigner: boolean;
```
Comment (nqhoan-axonivy) [inline marketplace-ui/src/app/shared/models/criteria.model.ts:10]: please rename the 'isRestDesigner' to is 'isRESTClientEditor'

**Review by nqhoan-axonivy — CHANGES_REQUESTED** [review]

**Review by tvtphuc-axonivy — CHANGES_REQUESTED** [review]

File: marketplace-ui/src/app/modules/home/home.component.spec.ts [hunk 9]
```diff
@@ -7,18 +7,27 @@ import {
   withInterceptorsFromDi
 } from '@angular/common/http';
 import { TranslateModule, TranslateService } from '@ngx-translate/core';
+import { ActivatedRoute } from '@angular/router';
+import { of } from 'rxjs';
 
 describe('HomeComponent', () => {
   let component: HomeComponent;
   let fixture: ComponentFixture<HomeComponent>;
+  let activatedRoute: ActivatedRoute;
```
Comment (tvtphuc-axonivy) [inline marketplace-ui/src/app/modules/home/home.component.spec.ts:16]: Please remove this line because you didn't use it

**Review by tvtphuc-axonivy — CHANGES_REQUESTED** [review]

**Review by tvtphuc-axonivy — APPROVED** [review]
LGTM

**Pull request merged**